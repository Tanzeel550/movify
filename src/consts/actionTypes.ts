export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MoviesActionReturnType = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type MoviesActionParamsType = {
  id?: string;
  movie?: object;
  movies?: object;
};

export type SEARCH_TYPE = {
  text: string;
};

export type ErrorType = {
  message: string;
  title: string;
};

export type ErrorActionsReturnType = {
  type: 'SET_ERROR' | 'CLEAR_ERROR';
  message?: string;
  title?: string;
};
