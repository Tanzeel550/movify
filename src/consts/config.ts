const OMBD_API_KEY = process.env.REACT_APP_OMDB_API_KEY;
export const SEARCH_MOVIE_BY_TEXT = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&s=`;
export const GET_MOVIE_BY_TITLE = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&t=`;
export const GET_MOVIE_BY_IMDB_ID = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&i=`;

export const PROMISE_AREAS = {
  SEARCH_MOVIE_BY_TEXT: 'SEARCH_MOVIE_BY_TEXT',
  GET_MOVIE_BY_TITLE: 'GET_MOVIE_BY_TITLE',
  GET_ALL_MOVIES: 'GET_ALL_MOVIES',
  LOAD_PAGE: 'LOAD_PAGE',
};

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const TEST_MOVIES_SRC = 'movies_test';
