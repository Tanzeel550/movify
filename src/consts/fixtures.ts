import { FireDBMovieItem } from '../types/APITypes';

export const movies: FireDBMovieItem[] = [
  {
    id: '1234567',
    name: 'The Dark Knight',
    watched: true,
    dateWatched: 234567890,
    whatYouLearnt: 'The world is as good as we let it to be',
    Poster: 'https://image.com',
    createdAt: 9876543,
  },
  {
    id: '4567890',
    name: 'GodFather',
    watched: false,
    dateWatched: 98765434567,
    whatYouLearnt: 'You can act like a man what is the matter with you?',
    Poster: 'https://godfather.com',
    createdAt: 987654567890,
  },
];

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const email = 'unknownPerson123@uet.com';
export const password = 'JonSnow';
