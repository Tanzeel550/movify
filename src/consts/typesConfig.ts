export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MOVIES_ACTION = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type SEARCH_TYPE = {
  text: string;
};

export type MOVIE_ID = {
  id?: string;
  movie?: object;
  movies?: object;
};
