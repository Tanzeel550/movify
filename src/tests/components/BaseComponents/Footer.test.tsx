import { shallow, ShallowWrapper } from 'enzyme';
import Footer, {
  SkillsComponent,
} from '../../../components/BaseComponents/Footer';
import React from 'react';
import { skills1 } from '../../../consts/fixtures';

test('should render footer correctly', () => {
  const wrapper: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  > = shallow(<Footer />);
  expect(wrapper).toMatchSnapshot();
});

test('should render SkillsComponent correctly', () => {
  const wrapper = shallow(<SkillsComponent skills={skills1} />);
  expect(wrapper).toMatchSnapshot();
});

export {};
