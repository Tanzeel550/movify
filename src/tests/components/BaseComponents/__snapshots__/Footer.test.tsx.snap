// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SkillsComponent correctly 1`] = `ShallowWrapper {}`;

exports[`should render footer correctly 1`] = `ShallowWrapper {}`;
